sdk/core/azure-core/src/main/java/com/azure/core/http/policy/RetryPolicy.java

```diff
@@ -190,8 +190,9 @@ private HttpResponse attemptSync(HttpPipelineCallContext context, HttpPipelineNe
                 logRetryWithError(LOGGER.atVerbose(), tryCount, "Error resume.", err);
                 try {
                     Thread.sleep(retryStrategy.calculateRetryDelay(tryCount).toMillis());
-                } catch (InterruptedException ie) {
-                    throw LOGGER.logExceptionAsError(new RuntimeException(ie));
+                } catch (InterruptedException ex) {
+                    err.addSuppressed(ex);
+                    throw LOGGER.logExceptionAsError(err);
```

Comment:
Change in design here where the outer caught exception (from the request failing) now adds the `InterruptedException` as a suppressed exception and is thrown itself. This makes sure that the exception retried isn't lost.